### ISSUE TITLE
I use the DBN train the data but i always get the  same result

### ISSUE THREAD

**Issue by abner2015** [deeplearning4j/deeplearning4j#1377]
`public class DBNEntity {

    private static Logger log = LoggerFactory.getLogger(DBNEntity.class);

    public static void main(String[] args) throws Exception {
    	String filePath = new ClassPathResource("testresult.txt").getFile().getAbsolutePath();
        RecordReader recordReader = new CSVRecordReader(0, " ");
        recordReader.initialize(new FileSplit(new File(filePath)));
        //reader,label index,number of possible labels
        org.nd4j.linalg.dataset.api.iterator.DataSetIterator iter = new RecordReaderDataSetIterator(recordReader, 400, 2);

        // Customizing params
        Nd4j.MAX_SLICES_TO_PRINT = 10;
        Nd4j.MAX_ELEMENTS_PER_SLICE = 10;
        
        final int numInputs = 400;
        int outputNum = 2;
        int iterations = 5;
        int seed = 12;
        int listenerFreq = iterations;
    	
    	final int numRows = 400;//
        final int numColumns = 2;
        
        int numSamples = 60000;
        int batchSize = 100;
       

        log.info("Load data....");
        //DataSetIterator iter = new MnistDataSetIterator(batchSize, numSamples, true);
        DataSet next = iter.next();
        next.normalizeZeroMeanZeroUnitVariance();
        
        
        log.info("Build model....");
        MultiLayerConfiguration conf = new NeuralNetConfiguration.Builder()
                .seed(seed)
                .gradientNormalization(GradientNormalization.ClipElementWiseAbsoluteValue)
                .gradientNormalizationThreshold(1.0)
                .iterations(iterations)
                .momentum(0.5)
                .momentumAfter(Collections.singletonMap(3, 0.9))
                .optimizationAlgo(OptimizationAlgorithm.CONJUGATE_GRADIENT)
                .list(5)
                .layer(0, new RBM.Builder().nIn(400).nOut(3)
                        .weightInit(WeightInit.XAVIER).lossFunction(LossFunction.RMSE_XENT)
                        .visibleUnit(RBM.VisibleUnit.BINARY)
                        .hiddenUnit(RBM.HiddenUnit.BINARY)
                        .build())
                .layer(1, new RBM.Builder().nIn(3).nOut(2)
                        .weightInit(WeightInit.XAVIER).lossFunction(LossFunction.RMSE_XENT)
                        .visibleUnit(RBM.VisibleUnit.BINARY)
                        .hiddenUnit(RBM.HiddenUnit.BINARY)
                        .build())
                .layer(2, new RBM.Builder().nIn(2).nOut(2)
                        .weightInit(WeightInit.XAVIER).lossFunction(LossFunction.RMSE_XENT)
                        .visibleUnit(RBM.VisibleUnit.BINARY)
                        .hiddenUnit(RBM.HiddenUnit.BINARY)
                        .build())
                .layer(3, new RBM.Builder().nIn(2).nOut(2)
                        .weightInit(WeightInit.XAVIER).lossFunction(LossFunction.RMSE_XENT)
                        .visibleUnit(RBM.VisibleUnit.BINARY)
                        .hiddenUnit(RBM.HiddenUnit.BINARY)
                        .build())
                .layer(4, new OutputLayer.Builder(LossFunction.NEGATIVELOGLIKELIHOOD).activation("softmax")
                        .nIn(2).nOut(outputNum).build())
                .pretrain(true).backprop(false)
                .build();
      //run the model
        MultiLayerNetwork model = new MultiLayerNetwork(conf);
        model.init();
        model.setListeners(Arrays.asList((IterationListener) new ScoreIterationListener(listenerFreq)));
       //model.setListeners(new HistogramIterationListener(1));

        next.normalizeZeroMeanZeroUnitVariance();
        next.shuffle();
        //split test and train
        SplitTestAndTrain testAndTrain = next.splitTestAndTrain(0.7);
        model.fit(testAndTrain.getTrain());
        //evaluate the model
        Evaluation eval = new Evaluation(5);
        DataSet test = testAndTrain.getTest();
        INDArray output = model.output(test.getFeatureMatrix());
        eval.eval(test.getLabels(), output);
        
        log.info(eval.stats());
        //save the model
        //DefaultModelSaver saveModel1 = new DefaultModelSaver();
        //saveModel1.save(model);
      //Write the network parameter`

**Comment by abner2015** [comment]
testresult.txt data

**Comment by abner2015** [comment]
[Uploading testresult.txt…]()

**Comment by nyghtowl** [comment]
@abner2015 you are using a seed so the same result is expected
